components/blocks/SearchBar/SearchBar.vue

```diff
@@ -1,25 +1,31 @@
 <script setup lang="ts">
-import { ref } from 'vue'
+import { ref, Ref } from 'vue'
 import BaseButton from '@/components/elements/buttons/BaseButton/BaseButton.vue'
 import BaseInput from '@/components/elements/inputs/BaseInput/BaseInput.vue'
 
-const inputValue = ref<string>('')
-// TODO: Find a fix for this prettier issue
-const emit = defineEmits<{(e: 'search', value: string): void}>() // prettier-ignore
+interface SearchBarState {
+  input: Ref<string>
+}
+
+const state: SearchBarState = {
+  input: ref<string>(''),
```

Comment:
Is it necessary to have `<string>` when having the empty string in the ref function will have TS implicitly know that it's of type `Ref(string)`

Reply:
No, I don't think it's necessary at all. Just did it out of habit. I'll remove them